src/eolab/rastertools/processing/sliding.py

```diff
@@ -58,17 +58,16 @@ def compute_sliding(input_image: str, output_image: str, rasterprocessing: Raste
 
             # dtype of output data
             dtype = rasterprocessing.dtype or rasterio.float32
-
             # check band index and handle all bands options (when bands is an empty list)
             if bands is None or len(bands) == 0:
                 bands = src.indexes
             elif min(bands) < 1 or max(bands) > src.count:
                 raise ValueError(f"Invalid bands, all values are not in range [1, {src.count}]")
 
             # setup profile for output image
-            profile.update(driver='GTiff',
+            profile.update(driver='GTiff', compress='lzw',
                            blockxsize=blockxsize, blockysize=blockysize, tiled=True,
-                           dtype=dtype, nodata=rasterprocessing.nodata or src.nodata,
+                           dtype=np.int8, nodata=rasterprocessing.nodata or src.nodata,
```

Comment:
you should modify the rasterprocessing attribute, and the default value above.

Reply:
> you should modify the rasterprocessing attribute, and the default value above.

So I tried to modify rasterprocessing attribute at first but dtype is used in the `process_map` from tqdm and it requires a 'same_kind' cast, so a float32 or float64 types allowed only. I am working on a solution.

> Isn't dtype=bool working?

bool is not working but I have just found out `dtype=np.ubyte or rasterio.ubyte` + `nbits=1` in profile is working and computes a proper output ever lighter than with np.int8 ( np.int8 -> 0.77MB; np.ubyte -> 0.55MB). I will add `compression` and `nbits` to rasterprocessing attributes ('nbits' might not be required, I'll check).

Reply:
Do you know why dtype=bool is not working? It should with Numpy. Does it comes from Numpy version or from another part of the code?

Reply:
The exact error is `AttributeError: 'bool' object has no attribute 'dtype'`. As I understood GDAL does not support a true 1-bit dtype and so the only way to do so would be to set a 8-bits dtype (such as np.int8 or np.ubyte which is also a 8-bits dtype) and to pass the NBITS=1 creation option to tell the underlying GDAL GeoTIFF driver to create a one bit file

Reply:
From where does come this error, do you hava astack trace?

Reply:
Yes I have a trace. After updating `profile `with `dtype=bool` , I got : 
`Traceback (most recent call last):
  File "test_rastertools.py", line 321, in <module>
    main(
  File "test_rastertools.py", line 134, in main
    tool.process_file(dem_path)
  File "/work/scratch/data/erblanm/QTIS/Theia/github_rastertools/rastertools/src/eolab/rastertools/hillshade.py", line 160, in process_file
    compute_sliding(
  File "/work/scratch/data/erblanm/QTIS/Theia/github_rastertools/rastertools/src/eolab/rastertools/processing/sliding.py", line 73, in compute_sliding
    with rasterio.open(output_image, "w", **profile):
  File "/work/scratch/env/erblanm/venv_lis_3.8_rastertools/lib/python3.8/site-packages/rasterio/env.py", line 451, in wrapper
    return f(*args, **kwds)
  File "/work/scratch/env/erblanm/venv_lis_3.8_rastertools/lib/python3.8/site-packages/rasterio/__init__.py", line 221, in open
    if dtype and not check_dtype(dtype):
  File "/work/scratch/env/erblanm/venv_lis_3.8_rastertools/lib/python3.8/site-packages/rasterio/dtypes.py", line 142, in check_dtype
    elif callable(dt) and str(dt().dtype) in dtype_rev:
AttributeError: 'bool' object has no attribute 'dtype'`

Reply:
OK, I see, this is not a Numpy problem but a rasterio/GDAL one, you are correct. So your proposed solution looks good!